lib/lattice_field.cpp

```diff
@@ -514,15 +648,13 @@ namespace quda {
 
   void *LatticeField::remoteFace_r() const { return ghost_recv_buffer_d[bufferIndex]; }
 
-  void LatticeField::setTuningString() {
-    char vol_tmp[TuneKey::volume_n];
-    int check  = snprintf(vol_string, TuneKey::volume_n, "%d", x[0]);
-    if (check < 0 || check >= TuneKey::volume_n) errorQuda("Error writing volume string");
-    for (int d=1; d<nDim; d++) {
-      strcpy(vol_tmp, vol_string);
-      check = snprintf(vol_string, TuneKey::volume_n, "%sx%d", vol_tmp, x[d]);
-      if (check < 0 || check >= TuneKey::volume_n) errorQuda("Error writing volume string");
-    }
+  void LatticeField::setTuningString()
+  {
+    std::stringstream vol_ss;
```

Comment:
Now that we switch to string we should use fmt here. Happy to delay that but we should add an issue then.

Reply:
Ok, I'll leave it to you file the issue on this.  😄